stores/flatten.go

```diff
@@ -60,6 +61,23 @@ func Flatten(in map[string]interface{}) map[string]interface{} {
 	return newMap
 }
 
+func FlattenMetadata(md Metadata) (map[string]interface{}, error) {
+	var mdMap map[string]interface{}
+	jsonBytes, err := json.Marshal(md)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(jsonBytes, &mdMap)
+	if err != nil {
+		return nil, err
+	}
+
+	flat := Flatten(mdMap)
+	encodeNonStrings(flat)
+	encodeNewLines(flat)
+	return flat, nil
```

Comment:
IMO the calls to `encodeNonStrings` and `encodeNewLines` should not be part of this function. They should be called from the stores directly. (Same for the reverse function.)

I can imagine that in the future, some stores might need some of these transformations, but not all of them. Having to create new wrappers that are mostly copy'n'paste of this function (and the reverse function) sounds unnecessarly complex to me.

Reply:
Done.  I moved the encode/decode NonStrings/NewLines calls out of this function